### PR TITLE
PWX-35851: Changes to actionCR definition for accommodating failover and failback actions.

### PR DESCRIPTION
**Pull request by diptiranjanpx**
Signed-Off-By: Diptiranjan

**What type of PR is this?**
api-change

**What this PR does / why we need it**:
ActionCR changes with new failover and failback operations.

**Does this PR change a user-facing CRD or CLI?**:

**Is a release note needed?**:

**Does this change need to be cherry-picked to a release branch?**:

### CODE REVIEW

File: pkg/apis/stork/v1alpha1/action.go [hunk 1]
```diff
@@ -34,27 +34,100 @@ type Action struct {
...
+// ActionSummary lists the summary of the action
+type ActionSummary ActionSummaryItem
+
+// ActionSummary has summary for each action type
+type ActionSummaryItem struct {
+	FailoverSummaryItem []*FailoverSummary `json:"failoverSummary,omitempty"`
+	FailbackSummaryItem []*FailbackSummary `json:"failbackSummary,omitempty"`
+}
+
+type FailoverSummary struct {
```
Comment (olavangad-px) [inline pkg/apis/stork/v1alpha1/action.go:124]: json names are incorrect for FailoverSummary and FailbackSummary

Comment (diptiranjanpx) [reply pkg/apis/stork/v1alpha1/action.go:124]: addressed.

**Review by olavangad-px — COMMENTED** [review]
LGTM

**Review by olavangad-px — APPROVED** [review]

File: pkg/apis/stork/v1alpha1/action.go [hunk 2]
```diff
@@ -34,27 +34,100 @@ type Action struct {
...
-	ActionType ActionType `json:"actionType"`
+	ActionType      ActionType       `json:"actionType"`
+	ActionParameter *ActionParameter `json:"actionParameter"`
 }
 
 // ActionType lists the various actions that can be performed
 type ActionType string
 
 const (
 	// to start apps on destination cluster
```
Comment (pp511) [inline pkg/apis/stork/v1alpha1/action.go:45]: nit...We can update this note to reflect the next line correctly.

File: pkg/apis/stork/v1alpha1/action.go [hunk 3]
```diff
@@ -34,27 +34,100 @@ type Action struct {
...
+	ActionTypeNearSyncFailover = "nearsyncFailover"
```
Comment (pp511) [inline pkg/apis/stork/v1alpha1/action.go:46]: Shouldn't it be  "ActionType NearSyncFailover"

**Review by pp511 — APPROVED** [review]
lgtm

File: pkg/apis/stork/v1alpha1/action.go (hunk 3 above)
Comment (olavangad-px) [reply pkg/apis/stork/v1alpha1/action.go:46]: I guess it should be `ActionTypeNearSyncFailover ActionType = "nearsyncFailover"`

File: pkg/apis/stork/v1alpha1/action.go (hunk 2 above)
Comment (diptiranjanpx) [reply pkg/apis/stork/v1alpha1/action.go]: Done

File: pkg/apis/stork/v1alpha1/action.go (hunk 3 above)
Comment (diptiranjanpx) [reply pkg/apis/stork/v1alpha1/action.go]: Done

**Pull request merged**